Extend support to add video in Gallery block.

Ta5r opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
- Enables users to add video files inside the core/gallery block.

<!-- Link this PR to its associated issue with an appropriate keyword: Closes, See, Follow up to, etc. -->
Closes <!-- #ISSUE-NUMBER or URL --> https://github.com/WordPress/gutenberg/issues/59572

<!-- In a few words, what is the PR actually doing? -->

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
- Previously, selecting a video in the Gallery block was not possible.
- This broke the user experience, who wanted to include videos alongside the photos.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
- Adds Video to be supported in Gallery block.
- When a video is detected, creates a core/video block with the correct id, src, and caption.
- Ensures seamless embedding of videos in the Gallery block, just like images.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

1. Add a Gallery block.
2. Click Add Media and select both an image and a video file. 
3. Observe: 
  - Images render directly. 
  - Videos now render as playable video blocks inside the gallery.

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

- Focus on the Gallery block using Tab.
- Use Enter/Return to open the media library or upload panel.
- Navigate via Tab/Arrow keys to select video files and press Enter.
- Once inserted, use Tab to navigate between inner blocks.
- Confirm that the video player is focusable and accessible.

## Screenshots or screencast <!-- if applicable -->



<!-- If you would like to upload screenshots, feel free to use the table below when it is useful to show the difference between before and after the change. -->

https://www.loom.com/share/c27db19969b14e66abe8c470e5016765?sid=83dcbb4c-567b-401d-8808-f1fa20d6cae8

Ta5r committed:
feat:Support video in Gallery block by creating core/video blocks.

Mamaduka commented on packages/block-library/src/gallery/edit.js:
```diff
@@ -541,7 +544,7 @@ export default function GalleryEdit( props ) {
 
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
-		defaultBlock: DEFAULT_BLOCK,
+		allowedBlocks: [ 'core/image', 'core/video' ],
 		directInsert: true,
```
Let's define `allowedBlocks` as a constant outside the component, since it's a static value - `const ALLOWED_BLOCKS = [ 'core/image', 'core/video' ]`.

The `directInsert` option can be removed now. It's only useful when inner blocks only allow a single block type.

Mamaduka reviewed (commented):
Thanks for contributing, @Ta5r!

I think adding support for the Video blocks would require more than just changing allowed block types. Most of the internal logic of the Gallery block is heavily image-focused; this needs to be adapted to account for both types.

I noticed that you're a first-time contributor. Of course, you're welcome to contribute in whatever way you want, but my advice is that this particular task might be better for an experienced contributor because of the size, scope and history.

Mamaduka commented:
Hi, @Ta5r

Would you like to continue working on this enhancement?

Ta5r committed:
refactor: inner blocks config for clarity

- Move allowedBlocks to a top-level constant (ALLOWED_BLOCKS)
- Remove unnecessary  property
This improves readability and aligns with Gutenberg coding standards.
